src/query/execute_statement.c

```diff
@@ -17775,13 +17775,21 @@ do_alter_synonym (PARSER_CONTEXT * parser, PT_NODE * statement)
   else
     {
       /* PT_SYNONYM_TARGET_NAME (statement) != NULL */
+      if (PT_SYNONYM_DBLINKED (statement))
+	{
+	  sprintf (target_name_buf, "%s", (PT_NAME_ORIGINAL (PT_SYNONYM_TARGET_NAME (statement))));
```

Comment:
The reason for writing to target_name_buf is to change it to lower case.
If target_name doesn't need to be changed to lowercase, we can set the pointer directly to target_name without writing target_name_buf.

But doesn't the dblink's target_name really need to be changed to lower case?

Reply:
but in case of not dblinked, it uses target_name_buf.
I'd like to keep the logic for readability.